builder/insert_test.go

```diff
@@ -4,212 +4,294 @@ import (
-	is := require.New(t)
-
-	// With columns
+var inserttests = []BuilderTest{
 	{
-		query := loukoum.Insert("table").Columns("a", "b", "c")
-		is.Equal("INSERT INTO table (a, b, c)", query.String())
-	}
-
-	// Without columns
+		Name:       "Columns With columns",
+		Builder:    loukoum.Insert("table").Columns("a", "b", "c"),
+		String:     "INSERT INTO table (a, b, c)",
+		Query:      "INSERT INTO table (a, b, c)",
+		NamedQuery: "INSERT INTO table (a, b, c)",
+	},
 	{
-		query := loukoum.Insert("table")
-		is.Equal("INSERT INTO table", query.String())
-	}
-}
-
-func TestInsert_Values(t *testing.T) {
-	is := require.New(t)
-
-	// With columns
+		Name:       "Columns Without columns",
+		Builder:    loukoum.Insert("table"),
+		String:     "INSERT INTO table",
+		Query:      "INSERT INTO table",
+		NamedQuery: "INSERT INTO table",
+	},
 	{
-		query := loukoum.
-			Insert("table").
+		Name: "Values With columns",
+		Builder: loukoum.Insert("table").
 			Columns("a", "b", "c").
-			Values([]string{"va", "vb", "vc"})
-
-		is.Equal("INSERT INTO table (a, b, c) VALUES ('va', 'vb', 'vc')", query.String())
-	}
-	{
-		query := loukoum.
-			Insert("table").
-			Columns("a", "b", "c").
-			Values("va", "vb", "vc")
-
-		is.Equal("INSERT INTO table (a, b, c) VALUES ('va', 'vb', 'vc')", query.String())
-	}
-
-	// Without columns
-	{
-		query := loukoum.
+			Values([]string{"va", "vb", "vc"}),
+		String:     "INSERT INTO table (a, b, c) VALUES ('va', 'vb', 'vc')",
+		Query:      "INSERT INTO table (a, b, c) VALUES ($1, $2, $3)",
+		NamedQuery: "INSERT INTO table (a, b, c) VALUES (:arg_1, :arg_2, :arg_3)",
+		Args:       []interface{}{"va", "vb", "vc"},
+		NamedArgs: map[string]interface{}{
+			"arg_1": "va",
+			"arg_2": "vb",
+			"arg_3": "vc",
+		},
+	},
+	{
+		Name: "Values Without columns",
+		Builder: loukoum.
 			Insert("table").
-			Values([]string{"va", "vb", "vc"})
-
-		is.Equal("INSERT INTO table VALUES ('va', 'vb', 'vc')", query.String())
-	}
-	{
-		query := loukoum.
-			Insert("table").
-			Values("va", "vb", "vc")
-
-		is.Equal("INSERT INTO table VALUES ('va', 'vb', 'vc')", query.String())
-	}
-
-	// With raw values
-	{
-		query := loukoum.
-			Insert("table").
-			Columns("email", "enabled", "created_at").
-			Values("tech@ulule.com", true, loukoum.Raw("NOW()"))
-
-		is.Equal("INSERT INTO table (email, enabled, created_at) VALUES ('tech@ulule.com', true, NOW())", query.String())
-	}
-}
-
-func TestInsert_OnConflict(t *testing.T) {
-	is := require.New(t)
-
-	// Do nothing without target
-	{
-		query := loukoum.
+			Values([]string{"va", "vb", "vc"}),
+		String:     "INSERT INTO table VALUES ('va', 'vb', 'vc')",
+		Query:      "INSERT INTO table VALUES ($1, $2, $3)",
+		NamedQuery: "INSERT INTO table VALUES (:arg_1, :arg_2, :arg_3)",
+		Args:       []interface{}{"va", "vb", "vc"},
+		NamedArgs: map[string]interface{}{
+			"arg_1": "va",
+			"arg_2": "vb",
+			"arg_3": "vc",
+		},
+	},
+	{
+		Name: "OnConflict Do nothing without target",
```

Comment:
I don't understand.

Reply:
If you could add a simple test case like `Values With mixed type` before the `OnConflict Do nothing without target` it would be perfect.

```
query := loukoum.
     Insert("table").
     Columns("email", "enabled", "created_at").
     Values("tech@ulule.com", true, loukoum.Raw("NOW()"))
```

I hope it was clear this time.